src/backend/renderer/gles2/mod.rs

```diff
@@ -1290,24 +1290,24 @@ impl Frame for Gles2Frame {
         let damage = damage
             .iter()
             .map(|rect| {
+                let src = src.size.to_f64();
                 let rect = rect.to_f64();
 
                 let rect_constrained_loc = rect
                     .loc
-                    .constrain(Rectangle::from_extemities((0f64, 0f64), dest.size.to_point()));
-                let rect_clamped_size = rect.size.clamp(
-                    (0f64, 0f64),
-                    (dest.size.to_point() - rect_constrained_loc).to_size(),
-                );
+                    .constrain(Rectangle::from_extemities((0f64, 0f64), src.to_point()));
```

Comment:
Do I understand correctly that the change from using `dest.size` to `src.size` is the mistake that is being fixed?

Reply:
Correct. The damage is relative to the buffer, so in cases where dest != src, the clamping may be wrong.